### ISSUE TITLE
MSSQL module has an undocumented username and password length limit

### ISSUE THREAD

**Issue by pv2b** [vanhauser-thc/thc-hydra#814]
The MSSQL module has an undocumented and silent username and password limit of 30 characters.

See here: https://github.com/vanhauser-thc/thc-hydra/blob/master/hydra-mssql.c#L68-L71

It just chops the password off without any warning to the user. I've not yet tried recompiling the code with a higher limit (maybe there are other protocol reasons for this limit?) but that might be worth trying for anyone who needs to explore brute forcing characters longer than 30 characters on MSSQL.

**Comment by vanhauser-thc** [comment]
maybe that comes from a time when that was the max password length? maybe it still is? I don't know, I didn't write the module :)

**Comment by pv2b** [comment]
Thank you for the feedback. Can you clarify what you meant though, did Hydra ever have a max password length of 30? Or are you suspecting maybe early versions of MSSQL had such a password limit?

**Comment by vanhauser-thc** [comment]
it could be that the limit was in early versions of the mssql database. or still is. I dont know.
I don't think there is a length limit for passwords in hydra itself. the modules can restrict the lengths themselves though like the mssql one.

**Comment by Goofygiraffe06** [comment]
```
 [24 Jun 2014 14:53] Jon Stephens

Fix documented in the 5.7.5 changelog, as follows:

    The maximum length that can be used for the password in a CHANGE
    MASTER TO statement is 32 characters. Previously, when a longer
    password was employed, any excess length was silently truncated
    by the server. Now when the password's length exceeds 32
    characters, CHANGE MASTER TO fails with an error.
  ```
  
  This is a documented fix from 2014 from [MySQL](https://bugs.mysql.com/bug.php?id=43439), where the server would silently truncate the password when it's length is greater than 32 characters. I will do a PR with an error message stating that the its truncating passwords who length is greater than 32 characters :D 
  
  Thanks,
  Goofygiraffe06

**Comment by motypi** [comment]
Yeah the issue is the version of TDS being used by the module, v4, which is still supported. I will try to update the module to use v7.
